packages/graphql/src/classes/utils/asserts-indexes-and-constraints.ts

```diff
@@ -83,47 +85,69 @@ async function createIndexesAndConstraints({ nodes, session }: { nodes: Node[];
         };
     });
 
-    nodes.forEach((node) => {
-        if (node.fulltextDirective) {
-            node.fulltextDirective.indexes.forEach((index) => {
-                // TODO: remove indexName assignment and undefined check once the name argument has been removed.
-                const indexName = index.indexName || index.name;
+    return existingIndexes;
+}
+
+async function createIndexesAndConstraints({
+    schemaModel,
+    session,
+}: {
+    schemaModel: Neo4jGraphQLSchemaModel;
+    session: Session;
+}) {
+    const constraintsToCreate = await getMissingConstraints({ schemaModel, session });
+    const indexesToCreate: { indexName: string; label: string; properties: string[] }[] = [];
+
+    const existingIndexes = await getExistingIndexes({ session });
+    const indexErrors: string[] = [];
+
+    for (const entity of schemaModel.concreteEntities) {
+        if (entity.annotations.fulltext) {
+            entity.annotations.fulltext.indexes.forEach((index) => {
+                const indexName = index.indexName || index.name; // TODO remove indexName assignment and undefined check once the name argument has been removed.
                 if (indexName === undefined) {
                     throw new Error("The name of the fulltext index should be defined using the indexName argument.");
                 }
                 const existingIndex = existingIndexes[indexName];
                 if (!existingIndex) {
                     const properties = index.fields.map((field) => {
-                        const stringField = node.primitiveFields.find((f) => f.fieldName === field);
+                        const attributeAdapter = entity.findAttribute(field);
```

Comment:
not too important but this is not an AttributeAdapter type, so the naming is a bit confusing